apps/src/LayerFromPPM.cpp

```diff
@@ -182,21 +185,38 @@ auto main(int argc, char* argv[]) -> int
     line->setSamplingDirection(direction);
 
     // Layer texture
-    std::cout << "Generating layers..." << std::endl;
     vc::texturing::LayerTexture s;
     s.setVolume(volume);
     s.setPerPixelMap(ppm);
     s.setGenerator(line);
+
+    if (parsed["progress"].as<bool>()) {
+        vc::ReportProgress(s, "Generating layers:");
+        vc::Logger()->debug("Generating layers...");
+    } else {
+        vc::Logger()->info("Generating layers...");
+    }
+
     auto texture = s.compute();
 
-    std::cout << "Writing layers..." << std::endl;
     const auto numChars =
         static_cast<int>(std::to_string(texture.size()).size());
     fs::path filepath;
-    for (const auto [i, image] : enumerate(texture)) {
-        auto fileName = vc::to_padded_string(i, numChars) + "." + imgFmt;
-        filepath = outputPath / fileName;
-        vc::WriteImage(filepath, image, writeOpts);
+    if (parsed["progress"].as<bool>()) {
+        vc::Logger()->debug("Writing layers...");
+        for (const auto [i, image] :
+             vc::ProgressWrap(enumerate(texture), "Writing layers:")) {
+            auto fileName = vc::to_padded_string(i, numChars) + "." + imgFmt;
+            filepath = outputPath / fileName;
+            vc::WriteImage(filepath, image, writeOpts);
+        }
+    } else {
+        vc::Logger()->info("Writing layers...");
+        for (const auto [i, image] : enumerate(texture)) {
+            auto fileName = vc::to_padded_string(i, numChars) + "." + imgFmt;
+            filepath = outputPath / fileName;
+            vc::WriteImage(filepath, image, writeOpts);
+        }
```

Comment:
I don't really like this double implementation. For a quick fix, put this at the top:

```c++
template<class Iterable>
void WriteLayers(const Iterable& iterable, const fs::path& outDir, int pad, const std::string& fmt, const vc::WriteImageOpts& opts) {
    for (const auto [i, image] : iterable) {
        auto fileName = vc::to_padded_string(i, pad) + "." + fmt;
        auto filepath = outDir / fileName;
        vc::WriteImage(filepath, image, opts);
    }
}
```

Then call it like this:
```c++
WriteLayers(enumerate(texture), outputPath, numChars, imgFmt, writeOpts);
WriteLayers(vc::ProgressWrap(enumerate(texture)), outputPath, numChars, imgFmt, writeOpts);
```

Since you're doing the same thing in two programs, you can also throw that function in a header somewhere in app support. Or just copy to both .cpps. It's a workaround for now.

You'll have to fix a bug in the `enumerate` and `ProgressWrap` calls for it to compile. In `app_support/ProgressIndicator.hpp`, add `const` after `begin()` and `end()` on lines 184 and 197. In `core/util/Iteration.hpp`, do the same on lines 471 and 474.